concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/10 Use Another Machine for Testing.md

```diff
@@ -0,0 +1 @@
+If DataGrid or TreeList is complex or the Internet connection is slow or unstable, gray boxes can be displayed for a very long time. If gray boxes take too long to disappear, you need to replace a remote data source with a local data source and open your project on a different machine.
```

Comment:
Well, I don't believe it's wise to address components as "complex". We speak of content, actually.

```suggestion
If DataGrid or TreeList content is complex or the Internet connection is slow or unstable, gray boxes can be displayed for a very long time. If gray boxes take too long to disappear, you need to replace a remote data source with a local data source and open your project on a different machine.
```